compute/cker/include/cker/train/operation/BinaryArithmetic.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NNFW_CKER_TRAIN_OPERATION_BINARYARITHMETIC_H__
+#define __NNFW_CKER_TRAIN_OPERATION_BINARYARITHMETIC_H__
+
+#include "cker/Shape.h"
+#include "cker/eigen/Utils.h"
+#include "cker/operation/BroadcastTo.h"
+
+namespace nnfw
+{
+namespace cker
+{
+namespace train
+{
+enum class ArithmeticType
+{
+  kAdd,
+  kSub,
+  kMul,
+  kDiv,
+};
+
+template <typename T>
+void BinaryArithmeticGrad(const Shape &incoming_shape, const T *incoming_data,
+                          const Shape &lhs_grad_shape, T *lhs_grad_data,
+                          const Shape &rhs_grad_shape, T *rhs_grad_data,
+                          ArithmeticType arithmetic_type)
+{
+  BroadcastTo(incoming_shape, const_cast<T *>(incoming_data), lhs_grad_shape, lhs_grad_data);
+  BroadcastTo(incoming_shape, const_cast<T *>(incoming_data), rhs_grad_shape, rhs_grad_data);
+  switch (arithmetic_type)
+  {
+    case ArithmeticType::kAdd:
+      break;
+    case ArithmeticType::kSub:
+    case ArithmeticType::kMul:
+    case ArithmeticType::kDiv:
```

Comment:
Are other operations always required to broadcast?

Reply:
I was planning to work with broadcasted data (which, to my understanding, is simple copying) for other OPs.
Is there a better alternative?

Reply:
Well, if there is a way to calculate both broadcast and gradient at the same time, it would be great. But it is not related to this PR, so let's think about it later. :)